crates/sui-framework/sources/governance/sui_system.move

```diff
@@ -271,12 +273,27 @@ module sui::sui_system {
 
         // Verify the represented address is an active or pending validator, and the capability is still valid.
         let verified_cap = validator_set::verify_cap(&self.validators, cap, ACTIVE_OR_PENDING_VALIDATOR);
-        let validator = validator_set::get_validator_mut_with_verified_cap(&mut self.validators, &verified_cap);
+        let validator = validator_set::get_validator_mut_with_verified_cap(&mut self.validators, &verified_cap, false /* include_candidate */);
 
         validator::request_set_gas_price(validator, verified_cap, new_gas_price);
     }
 
-    /// A validator can call this entry function to set a new commission rate, updated at the end of the epoch.
+    /// This entry function is used to set new gas price for candidate validators
+    public entry fun set_candidate_validator_gas_price(
+        wrapper: &mut SuiSystemState,
+        cap: &UnverifiedValidatorOperationCap,
+        new_gas_price: u64,
+    ) {
+        let self = load_system_state_mut(wrapper);
+
+        // Verify the represented address is an active or pending validator, and the capability is still valid.
+        let verified_cap = validator_set::verify_cap(&self.validators, cap, ACTIVE_OR_PENDING_VALIDATOR);
```

Comment:
Copy pasta? We are setting candidate's gas price, not active or pending validator.

Reply:
This doesn't seem resolved? It should be ANY_VALIDATOR right?

Reply:
I don't know why it does not showing but it's in (unless I am going insane): https://github.com/MystenLabs/sui/pull/9044/files#diff-d9f8c41cba67e20d88edf44b94e60831b4d1613b6648be5b2a5e018664fc8e89R290